Assignment/Assignment/SampleData.cs

```diff
@@ -1,30 +1,45 @@
 ﻿using System;
+using System.IO;
 using System.Collections.Generic;
+using System.Linq;
 
 namespace Assignment
 {
     public class SampleData : ISampleData
     {
+
+        private const int FirstNameColumn = 1;
+        private const int LastNameColumn = 2;
+        private const int EmailColumn = 3;
+        private const int StreetAddressColumn = 4;
+        private const int CityColumn = 5;
+        private const int StateColumn = 6;
+        private const int ZipColumn = 7;
+
         // 1.
-        public IEnumerable<string> CsvRows => throw new NotImplementedException();
+        public IEnumerable<string> CsvRows => File.ReadAllLines(Path.Combine(Directory.GetParent(Directory.GetCurrentDirectory())!.Parent!.Parent!.Parent!.FullName, "Assignment" , "People.csv")).Skip(1);
+
 
         // 2.
-        public IEnumerable<string> GetUniqueSortedListOfStatesGivenCsvRows() 
-            => throw new NotImplementedException();
+        public IEnumerable<string> GetUniqueSortedListOfStatesGivenCsvRows() => CsvRows.Select(line=>line.Split(",")[StateColumn]).Distinct().OrderBy(state=>state);
+
 
         // 3.
-        public string GetAggregateSortedListOfStatesUsingCsvRows()
-            => throw new NotImplementedException();
+        public string GetAggregateSortedListOfStatesUsingCsvRows() =>  GetUniqueSortedListOfStatesGivenCsvRows().Aggregate((result, item) => result + ", " + item);
+
 
         // 4.
-        public IEnumerable<IPerson> People => throw new NotImplementedException();
+        // Ordered by State, City, ZipCode
+        public IEnumerable<IPerson> People => CsvRows.Select(line => line.Split(",")).OrderBy(line => line[StateColumn]).ThenBy(line => line[CityColumn]).ThenBy(line => line[ZipColumn])

```

Comment:
```suggestion
        public IEnumerable<IPerson> People
            => CsvRows
                .Select(line => line.Split(","))
                .OrderBy(line => line[StateColumn])
                .ThenBy(line => line[CityColumn])
                .ThenBy(line => line[ZipColumn])
                .Select(person => new Person(
                    person[FirstNameColumn],
                    person[LastNameColumn],
                    new Address(
                        person[StreetAddressColumn],
                        person[CityColumn],
                        person[StateColumn],
                        person[ZipColumn]
                    ),
                    person[EmailColumn]
                ));
```

For improved readability, I would suggest multi lining these functions.